x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/predicate/regex/RegexMatch.java

```diff
@@ -28,8 +29,11 @@ public DataType dataType() {
     }
 
     @Override
-    public boolean nullable() {
-        return field().nullable() && pattern != null;
+    public Nullability nullable() {
+        if (pattern == null) {
```

Comment:
I don't think the change here has the same outcome as the previous code?

Reply:
yep, but before the logic was wrong as LIKE/RLIKE return null if the pattern is null or if the value checked is null.